react/Table/Readme.md

```diff
@@ -70,59 +73,30 @@ const columns = [
   }
 ]
 
-initialState = { selectedItemsId: [] }
-
-const isSelectedItem = row => state.selectedItemsId.some(selectedItem => selectedItem === row.id)
-const addSelected = row => setState(prev => {
-    const arr = prev.selectedItemsId
-    arr.push(row.id)
-    return { selectedItemsId: arr }
-})
-const removeSelected = (row) => setState(prev => {
-    const arr = prev.selectedItemsId
-    arr.splice(arr.indexOf(row.id), 1)
-    return { selectedItemsId: arr }
-})
-const onSelect = row => {
-  if (isSelectedItem(row)) {
-    removeSelected(row)
-  } else {
-    addSelected(row)
-  }
-}
-const onSelectAll = rows => {
-  const ids = rows.map(row => row.id)
-  if (state.selectedItemsId.length === ids.length) {
-    setState({ selectedItemsId: [] })
-  } else {
-    setState({ selectedItemsId: ids })
-  }
-}
-
 const initialVariants = [{ grouped: false }]
 
-const makeGroups = () => ({ groupLabels: ['C', 'D', 'Others'], groupCounts: [1,1,12] })
+// Very basic usage only works when Dessert is sorted "asc"
+// Ideally you have to create a logic to create groups with sorted data
+const makeGroups = () => ({ groupLabels: ['C', 'D', 'E', 'Others'], groupCounts: [1,1,1,11] })
 
-/**
- * @param order
- * @param orderBy
- */
-const onSortChange = ({ order, orderBy }) => {
-  setState({ order, orderBy })
-}
+const ExampleTable = ({ variant, ...props }) => {
+  const { selectedItemsId, isSelectedItem, toggleSelectedItem, toggleSelectAllItems } = useSelection()
 
-<Variants initialVariants={initialVariants} screenshotAllVariants>
-  {variant => (
+  const onSortChange = ({ order, orderBy }) => {
+    console.info('order:', order, 'oderBy:', orderBy)
```

Comment:
we can keep the same setState example in the readme

Reply:
If think it was a bug to use it here. `setState` refers to `initialState` and there is no `order` neither `orderBy` in it. There is no purpose to use this callback in the doc as is, so I think just a console do the tricks